### PR TITLE
Fix Julia v0.5 deprecations

### PR DESCRIPTION
**Pull request by TotalVerb**
There aren't that many, and it seems they're pretty easily fixed.

### CODE REVIEW

**Comment by TotalVerb** [comment]
Shame on me for not testing on release.

File: src/JSON.jl [hunk 1]
```diff
@@ -292,7 +293,7 @@ end
 function parsefile{T<:Associative}(filename::AbstractString; dicttype::Type{T}=Dict, use_mmap=true)
     sz = filesize(filename)
     open(filename) do io
-        s = use_mmap ? UTF8String(Mmap.mmap(io, Vector{UInt8}, sz)) : readall(io)
+        s = use_mmap ? @compat(String(Mmap.mmap(io, Vector{UInt8}, sz))) : readall(io)
```
Comment (yuyichao) [inline src/JSON.jl]: I don't think the `@compat` is necessary here?

Comment (yuyichao) [inline src/JSON.jl]: And I believe you want `Compat.UTF8String` instead...

File: src/Parser.jl [hunk 2]
```diff
@@ -1,6 +1,7 @@
 module Parser #JSON
 
 using Compat
+import Compat: String
```
Comment (yuyichao) [inline src/Parser.jl]: This is not used.

File: src/JSON.jl [hunk 3]
```diff
@@ -3,6 +3,7 @@ isdefined(Base, :__precompile__) && __precompile__()
 module JSON
 
 using Compat
+import Compat: String
```
Comment (yuyichao) [inline src/JSON.jl]: This can be removed with the case below fixed.

File: REQUIRE [hunk 4]
```diff
@@ -1,2 +1,2 @@
 julia 0.3
-Compat 0.7.1
+Compat 0.7.16
```
Comment (yuyichao) [inline REQUIRE]: The version bump is actually not necessary but doesn't hurt either....

**Comment by TotalVerb** [comment]
Thanks a ton for the comments. If this passes CI then I'll squash the commits.

File: REQUIRE (hunk 4 above)
Comment (tkelman) [inline REQUIRE]: it is for `Compat.UTF8String`, isn't it?

**Comment by tkelman** [comment]
LGTM

File: REQUIRE (hunk 4 above)
Comment (yuyichao) [inline REQUIRE]: Only for suppressing warnings. It will work without.

**Pull request merged**